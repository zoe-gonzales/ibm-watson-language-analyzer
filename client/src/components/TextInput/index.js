import React from 'react';

const TextInput = ({
    name,
    placeholder,
    onChange
}) => {
    return (
        <div>
            <label htmlFor={name}>{placeholder}</label>
            <textarea id={name} className="materialize-textarea" onChange={onChange}></textarea>
        </div>
    )
}

export default TextInput;