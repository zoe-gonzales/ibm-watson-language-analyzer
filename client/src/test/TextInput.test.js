import React from 'react';
import TextInput from '../components/TextInput';
import renderer from 'react-test-renderer';

it('renders TextInput component correctly', () => {
    const name = 'textToSearch';
    const placeholder = 'Text to analyze...';
    const handleChange = e => {
        console.log(`Event: ${e.target}`);
        console.log('Text input field changed');
    };
    const component = renderer
    .create(<TextInput name={name} placeholder={placeholder} onChange={handleChange} />)
    .toJSON();
    expect(component).toMatchSnapshot();
});