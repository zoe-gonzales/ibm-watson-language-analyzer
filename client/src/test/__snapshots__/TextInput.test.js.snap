// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TextInput component correctly 1`] = `
<div>
  <label
    htmlFor="textToSearch"
  >
    Text to analyze...
  </label>
  <textarea
    className="materialize-textarea"
    id="textToSearch"
    onChange={[Function]}
  />
</div>
`;
